### ISSUE TITLE
Password field: Inconsistent autofocus behavior between the Post Editor and the Site Editor (no autofocus)

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52605]
### Description

Discovered while investigating https://github.com/WordPress/gutenberg/issues/52601

Auto-focusing input fields should be avoided for the reasons mentioned in https://github.com/WordPress/gutenberg/issues/52601. There's even a `jsx-a11y/no-autofocus` eslint rule in place that should never be disabled, unless there are very very good reasons to do that.

Besides that, initial hte autofocus behavior on the password field is inconsistent across the Post Editor and the Site Editor.

- When editing a post or page in the Post editor > Visibility > Password protected: focus stays on the radio button and the password input field is _not_ autofocused.
-  When editing a page in the Site editor > Pages > Any page > Status > Hide this page behind a password, the password input field is autofocused.

Screenshot, from left to right:
- Post editor, editing a post.
- Post editor, editing a page.
- Site editor, editing a page.

<img width="1005" alt="Screenshot 2023-07-13 at 16 19 17" src="https://github.com/WordPress/gutenberg/assets/1682452/19ab929b-2fce-42db-b30b-a096693c0359">

I'm not sure why the UI within the popover is different and I'm not sure why in the Site editor Visibility and Status properties are mixed together in the same UI. Besides that, the focus behavior is inconsistent and autofocus should be avoided.

Additionally, some wording is inconsistent, although used on different elements. E.g.:
- 'Use a secure password' vs. 'Enter a secure password'.
- 'Password protected' vs. 'Password'.
- 'Only those with the password can view this post.' vs. 'Hide this page behind a password'

### Step-by-step reproduction instructions

- Go to the Post editor, edit Visibility in the Settings sidebar.
- Click 'Password protected'.
- Observe the password input field appears but focus stays on the radio button you previously clicked.
- Go to the Site editor > Design > Pages > Edit any page.
- In the Settings sidebar, edit the Status.
- Click the toggle 'Hide this page behind a password'.
- Observe the password input field is autofocused.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Additionally, just noticed:
- In the Post editor, the password input field is of `type="text'` so that users can actually _see_ the password they enter.
- Instead, in the Site editor, the password input field is of `type="password'`. There is no way users can _see_ the password they are entering. While this should be reported and addressed di a separate issue, I wonder how this filed is any usable.

Screenshot:

<img width="689" alt="Screenshot 2023-07-13 at 16 49 36" src="https://github.com/WordPress/gutenberg/assets/1682452/eac3c188-91ad-4b22-b507-a09786511a49">
